src/components/editor/Store.tsx

```diff
@@ -13,13 +13,17 @@ export interface EditorStoreState<T> {
     // TODO (typing) : This needs typed. Using the T here made the checks in setSpecs break
     specs: any[] | null;
     setSpecs: (newVal: EditorStoreState<T>['specs']) => void;
+
+    serverUpdate: any | null;
+    setServerUpdate: (newVal: EditorStoreState<T>['serverUpdate']) => void;
```

Comment:
Initially I planned to keep a queue of updates and the user could clear them out. Then decided best start with just supporting one. This does create a possible edgecase of you missing a server update that happened _while_ handling the last server update.